Framework/Algorithms/src/SetUncertainties.cpp

```diff
@@ -120,32 +125,39 @@ void SetUncertainties::exec() {
   double valueToSet = resetOne ? 1.0 : getProperty("SetErrorTo");
   double valueToCompare = resetOne ? 0.0 : getProperty("IfEqualTo");
   int precision = getProperty("Precision");
-  double tolerance = resetOne ? 1E-10 : std::pow(10.0, precision * (-1));
+  double tolerance = resetOne ? 1E-10 : std::pow(10.0, -1. * precision);
 
   // Create the output workspace. This will copy many aspects from the input
   // one.
-  MatrixWorkspace_sptr outputWorkspace =
-      WorkspaceFactory::Instance().create(inputWorkspace);
 
-  // ...but not the data, so do that here.
+  const int64_t numHists =
+      static_cast<int64_t>(inputWorkspace->getNumberHistograms());
+  if (inputEventWorkspace) {
+    outputWorkspace = WorkspaceFactory::Instance().create(inputWorkspace);
```

Comment:
A comment pointing out that this copies the histogram representation over would be helpful